documentation/contributing/code/index.markdown

```diff
@@ -23,7 +23,9 @@ In addition MoveIt has some extra style preferences:
  - Avoid C-style functions such as ``FLT_EPSILON`` - instead use ``std::numeric_limits<double>::epsilon()``
  - Boost is an encouraged library when functionality is not available in the standard library
  - Prefer full variable names over short acryonms - e.g. ``robot_state_`` over ``rs_``
- - Deprecate functions using MOVEIT_DEPRECATED in [deprecation.h](https://github.com/ros-planning/moveit/blob/master/moveit_core/macros/include/moveit/macros/deprecation.h)
+ - Deprecate functions using C++14 [deprecated](https://en.cppreference.com/w/cpp/language/attributes/deprecated) attribute
```

Comment:
One could encourage users to add a message describing how to handle the situation as in the example I just stole from [here](https://josephmansfield.uk/articles/marking-deprecated-c++14.html):

`[[deprecated("use bar instead")]]
void foo() {}
`

Which would result in 

`
warning: 'foo' is deprecated: use bar instead [-Wdeprecated-declarations]
        foo();
        ^
note: 'foo' has been explicitly marked deprecated here
void foo() {}
     ^`

Reply:
Done (and cleaned up a lot of other stuff)